src/dioptra/restapi/v1/schemas.py

```diff
@@ -24,63 +24,110 @@
+            attribute="snapshot_id",
+            metadata=dict(description=f"ID for the underlying {name} snapshot."),
+            dump_only=True,
+        ),
+        "groupId": fields.Integer(
+            attribute="group_id",
+            metadata=dict(
+                description=f"ID of the Group that will own the {name} resource."
             ),
-            "createdOn": fields.DateTime(
-                attribute="created_on",
-                metadata=dict(
-                    description=f"Timestamp when the {name} resource was created."
-                ),
-                dump_only=True,
+            load_only=True,
+        ),
+        "group": fields.Nested(
+            GroupRefSchema,
+            attribute="group",
+            metadata=dict(description=f"Group that owns the {name} resource."),
+            dump_only=True,
+        ),
+        "user": fields.Nested(
+            UserRefSchema,
+            attribute="user",
+            metadata=dict(description=f"User that created the {name} resource."),
+            dump_only=True,
+        ),
+        "createdOn": fields.DateTime(
+            attribute="created_on",
+            metadata=dict(
+                description=f"Timestamp when the {name} resource was created."
             ),
-            "lastModifiedOn": fields.DateTime(
-                attribute="last_modified_on",
-                metadata=dict(
-                    description=f"Timestamp when the {name} resource was last modified."
-                ),
-                dump_only=True,
+            dump_only=True,
+        ),
+        "lastModifiedOn": fields.DateTime(
+            attribute="last_modified_on",
+            metadata=dict(
+                description=f"Timestamp when the {name} resource was last modified."
             ),
-            "tags": fields.Nested(
-                TagRefSchema,
-                attribute="tags",
-                metadata=dict(description="Tags associated with the {name} resource."),
-                many=True,
-                dump_only=True,
+            dump_only=True,
+        ),
+        "latestSnapshot": fields.Bool(
+            attribute="latest_snapshot",
+            metadata=dict(
+                description=f"Whether or not the {name} resource is the latest version."
             ),
-        }
-    )
+            dump_only=True,
+        ),
+        "tags": fields.Nested(
+            TagRefSchema,
+            attribute="tags",
+            metadata=dict(description="Tags associated with the {name} resource."),
+            many=True,
+            dump_only=True,
+        ),
+    }
+
+    if not snapshot:
+        schema.pop("snapshotId")
+        schema.pop("user")
+        schema.pop("createdOn")
+        schema.pop("lastModifiedOn")
+        schema.pop("latestSnapshot")
+
+    return Schema.from_dict(schema, name=f"{name}BaseSchema")
+
+
+def generate_base_resource_ref_schema(name: str, snapshot: bool) -> type[Schema]:
```

Comment:
The meaning of the `snapshot` parameter is slightly different in `generate_base_resource_schema` and this function. I noticed that in the `schema.py` file for the same entity (such as queues), we end up using `snapshot=True` for the former and `snapshot=False` for the latter, which might lead to confusion. I propose we change it to `keep_snapshot_id` here to make the intent clearer. (We can leave it as `snapshot` in the other function, it makes sense there).

```suggestion
def generate_base_resource_ref_schema(
    name: str, keep_snapshot_id: bool
) -> type[Schema]:
```

Reply:
made this change.

I also opted to make `keep_snapshot_id` default to `False` because I think the only place it will be `True` is in the `SnapshotRef` in `schemas.py`. It's use for the Ref types for each of the other endpoints should be `False`.

I kept `snapshot` without a default because we will have a mix of `Resource` and `ResourceSnapshot` types in the endpoints and I want it to be clear which it is.